### DISCUSSION TITLE
`Cookies.set()` should prepend a period to the domain

### DISCUSSION THREAD

**Discussion by ppena-LiveData** [encode/httpx#2833] [Ideas]
Since `httpx._models.Cookies` uses `http.cookiejar`, `Cookies.set()` should check if the passed-in domain has a leading period, and if not, it should prepend a period. When there is no leading period for the cookie's domain, then `cookiejar.CookieJar.clear()` cannot delete the cookie, and since prepending a period is not really intuitive, `httpx._models.Cookies` should do that for the user.

For those wondering, [cookiejar.CookieJar.clear()](https://github.com/python/cpython/blob/13a00078b81776b23b0b6add69b848382240d1f2/Lib/http/cookiejar.py#L1692) just does a `del self._cookies[domain][path][name]`, so the domain has to match exactly, and [this code that calls it](https://github.com/python/cpython/blob/13a00078b81776b23b0b6add69b848382240d1f2/Lib/http/cookiejar.py#L1535C13-L1535C32) is the thing that adds the period to the web response's cookie domain right before [CookieJar.clear()](https://github.com/python/cpython/blob/13a00078b81776b23b0b6add69b848382240d1f2/Lib/http/cookiejar.py#L1559C17-L1559C47) is called.

**Comment by lovelydinosaur** [comment]
Could you give us an example...

**Comment by ppena-LiveData** [comment]
Sorry, upon further review, it looks like it's really an `http.cookiejar` issue, so I'll file the bug with them instead.

https://github.com/python/cpython/issues/108931
